pkg/cmd/server/server.go

```diff
@@ -150,7 +150,12 @@ func (o *options) run(cmd *cobra.Command) error {
 		return errors.Trace(err)
 	}
 	// Drain the server before shutdown.
-	shutdownNotify := func() <-chan struct{} { return server.Drain(ctx) }
+	shutdownNotify := func() <-chan struct{} {
```

Comment:
shutdownNotify returns a channel, because it should be asynchronous. `InitSignalHandling` can abort drain when it receives a second signal.